Adapter/Algolia.php

```diff
@@ -90,140 +96,200 @@ public function __construct(
-            } catch (AlgoliaConnectionException $e) {
-                $useNative = true;
+            // If instant search is on, do not make a search query unless SEO request is set to 'Yes'
+            if (!$this->config->isInstantEnabled($storeId) || $this->config->makeSeoRequest($storeId)) {
+                $documents = $this->getDocumentsFromAlgolia($algoliaQuery, $storeId);
             }
-        } else {
-            $useNative = true;
-        }
 
-        if ($useNative) {
-            $nativeQueryData = $this->getNativeQueryData($request);
-            $documents = $nativeQueryData['documents'];
-            $table = $nativeQueryData['table'];
+            $apiDocuments = array_map([$this, 'getApiDocument'], $documents);
+            $table = $temporaryStorage->storeApiDocuments($apiDocuments);
+        } catch (AlgoliaConnectionException $e) {
+            $this->nativeQuery($request);
         }
 
+        $aggregations = $this->aggregationBuilder->build($request, $table, $documents);
         $response = [
             'documents' => $documents,
-            'aggregations' => $this->getAggregations($request, $table, $documents),
+            'aggregations' => $aggregations,
         ];
 
         return $this->responseFactory->create($response);
     }
 
-    /**
-     * Get native query documents
-     *
-     * @param  RequestInterface $request
-     *
-     * @return array
-     */
-    public function getNativeQueryData($request)
+    private function nativeQuery(RequestInterface $request)
     {
-        $query  = $this->mapper->buildQuery($request);
+        $query = $this->mapper->buildQuery($request);
         $temporaryStorage = $this->temporaryStorageFactory->create();
         $table = $temporaryStorage->storeDocumentsFromSelect($query);
+
         $documents = $this->getDocuments($table);
 
-        return [
+        $aggregations = $this->aggregationBuilder->build($request, $table, $documents);
+        $response = [
             'documents' => $documents,
-            'table' => $table,
+            'aggregations' => $aggregations,
         ];
+
+        return $this->responseFactory->create($response);
+    }
+
+    private function getApiDocument($document)
+    {
+        return $this->documentFactory->create($document);
     }
 
     /**
-     * Checks if Algolia is properly configured and enabled
+     * Get search result from Algolia
      *
-     * @param  int     $storeId
+     * @param string $algoliaQuery
+     * @param int $storeId
      *
-     * @return bool
+     * @return array
      */
-    private function isAllowed($storeId)
+    private function getDocumentsFromAlgolia($algoliaQuery, $storeId)
     {
-        return
-            $this->config->getApplicationID($storeId)
-            && $this->config->getAPIKey($storeId)
-            && $this->config->isEnabledFrontEnd($storeId)
-            && $this->config->makeSeoRequest($storeId);
+        $contextParams = [];
```

Comment:
In Algolia we call it a `searchParameters` rather than `contextParameters`. Can we keep that naming (even for further variables and methods) or did you choose `contextParameters` for a specific reason?